src/Microsoft.Diagnostics.Monitoring/Dumper.Windows.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Microsoft.Win32.SafeHandles;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    /// <summary>
+    /// TODO Share this with dotnet-dump
+    /// </summary>
+    internal static class Dumper
+    {
+        internal static Task CollectDumpAsync(Process process, string outputFile, NETCore.Client.DumpType type)
+        {
+            // We can't do this "asynchronously" so just Task.Run it. It shouldn't be "long-running" so this is fairly safe.
+            return Task.Run(() =>
+            {
+                // Open the file for writing
+                using (var stream = new FileStream(outputFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
```

Comment:
This Stream isn't async at the moment, you need to use the constructor that has more arguments 😄

Reply:
This was the original code for coredumps. Does MiniDumpWriteDump make use of overlapped io?

---
In reply to: [403434379](https://github.com/dotnet/diagnostics/pull/979#discussion_r403434379) [](ancestors = 403434379)

Reply:
Good question

Reply:
As far as I know MiniDumpWriteDump only allows synchronous invocation and writes to a file. It is possible that in the implementation details overlapped io is used, but there would be no way to leverage it without asking the OS team for a newer API. I assume it will be rare to collect more than one dump at a time so it doesn't seem too worrisome to spend a thread on it.

Reply:
Seems fine then. This is the most inefficient server doing IO 😁. Good thing it’s a side car and not multi-tenant 😬